keel-ec2-api/src/main/kotlin/com/netflix/spinnaker/keel/api/ec2/SecurityGroupRule.kt

```diff
@@ -24,18 +25,14 @@ sealed class SecurityGroupRule {
   enum class Protocol {
     ALL, TCP, UDP, ICMP
   }
-
-  open val isSelfReference: Boolean = false
 }
 
 data class ReferenceRule(
   override val protocol: Protocol,
-  val name: String? = null,
+  @Optional("defaults to the name of the security group the rule belongs to")
+  val name: String,
```

Comment:
`name` is non-nullable now. The `@Optional` annotation just tells the schema generator not to mark it as required.